FinancialTracker/SceneDelegate.swift

```diff
@@ -20,11 +20,16 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
         let window = UIWindow(windowScene: windowScene)
         let storyboard = UIStoryboard(name: "Main", bundle: nil)
         let entryVC = storyboard.instantiateViewController(withIdentifier: "EntryVC") as? EntryViewController
-        let mainVC = storyboard.instantiateViewController(withIdentifier: "BalanceVC") as? BalanceViewController
+        let balanceVC = storyboard.instantiateViewController(withIdentifier: "BalanceVC") as? BalanceViewController
+        let homeVC = storyboard.instantiateViewController(withIdentifier: "HomeVC") as? HomeViewController
         
-        FirebaseHandler.shared.checkAuthorisedState { success in
-            if success {
-                window.rootViewController = mainVC
+        FirebaseHandler.shared.checkAuthorisedState { user in
+            if user != nil {
```

Comment:
better use guard here.

Reply:
I can use guard but because i have to use return in the else statement i have to place these 2 times more:
`self.window = window
  self.window?.makeKeyAndVisible()`

Reply:
try using guard let or if let